exporter/datadogexporter/internal/translator/metrics_translator.go

```diff
@@ -152,6 +155,35 @@ func getBounds(p pdata.HistogramDataPoint, idx int) (lowerBound float64, upperBo
 	return
 }
 
+func (t *Translator) getSketchBuckets(name string, ts uint64, p pdata.HistogramDataPoint, delta bool, tags []string) sketches.SketchSeries {
+	as := &quantile.Agent{}
+	for j := range p.BucketCounts() {
+		lowerBound, upperBound := getBounds(p, j)
+		if math.IsInf(upperBound, 1) {
+			// Set it to the lower bound because of this mysterious comment:
+			// https://github.com/DataDog/datadog-agent/blob/7.30.1/pkg/aggregator/check_sampler.go#L107-L111
```

Comment:
It's probably because `InsertInterpolate` doesn't work with an infinite bound, right? So essentially, if I understand correctly, what this does is inserting the full count of that bucket into the receiving bucket that contains `lowerBound`.